flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -412,31 +418,22 @@ public void createPartition(ObjectPath tablePath, CatalogPartitionSpec partition
 		checkNotNull(partitionSpec);
 		checkNotNull(partition);
 
-		if (!tableExists(tablePath)) {
```

Comment:
I think it may not worth extract these to their own methods like `ensureTableExists` since it's such a simple logic and very clear to readers what happens.  When I read this code now, I have to find and check where the `ensureXxx` are and what they do.

Reply:
It's intended to reduce code duplication. And makes things easier if we want to, say, change exception message in the future.